### PR TITLE
Add a parse_token function

### PR DESCRIPTION
**Pull request by thesadru**
Allow the user to parse their bot token to get the user id and other data from it.

image

Sadly there's not much use for this feature as the clientuser is created right on start during login.

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/utils.py [hunk 1]
```diff
@@ -316,6 +317,32 @@ def oauth_url(
...
 
 
+def parse_token(token: str) -> Tuple[int, datetime.datetime, bytes]:
+    """Parse a token into its parts
+    
+    Returns 
+    
+    Parameters
+    -----------
+    token: :class:`token`
```
Comment (m1raynee) [inline disnake/utils.py:327]: do we have token class?
```suggestion
    token: :class:`str`
```

File: disnake/utils.py [hunk 2]
```diff
@@ -316,6 +317,32 @@ def oauth_url(
...
+    Returns
+    --------
+    Tuple[:class:`int`, :class:`datetime.datetime`, :class:`bytes`]
+        the bot's id, the time when the token was generated and the hmac.
+    """
+    parts = token.split('.')
+    
+    user_id = int(b64decode(parts[0]))
+    
+    timestamp = int.from_bytes(b64decode(parts[1] + '=='), 'big')
```
Comment (m1raynee) [inline disnake/utils.py:339]:
```suggestion
    timestamp = int.from_bytes(b64decode(parts[1] + '=='), 'big')
    timestamp += 1293840000  # Epoch used for tokens
```

**Review by m1raynee — CHANGES_REQUESTED** [review]
beep boop bep boop

File: disnake/utils.py (hunk 1 above)
Comment (thesadru) [reply disnake/utils.py:327]: Totes my bad, will fix it once I get the chance

File: disnake/utils.py (hunk 2 above)
Comment (thesadru) [reply disnake/utils.py:339]: Actually ain't a thing, the image seems to be outdated cause it works just fine without an epoch, idk why tho.

**Pull request merged**